.github/workflows/gpu_4090_full_benchmark.yml

```diff
@@ -95,6 +95,87 @@ jobs:
           -d @${{ env.RESULTS_FILENAME }} \
           ${{ secrets.SLAB_URL }}
 
+      - name: Slack Notification
+        if: ${{ always() }}
+        continue-on-error: true
+        uses: rtCamp/action-slack-notify@b24d75fe0e728a4bf9fc42ee217caa686d141ee8
+        env:
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MESSAGE: "Integer RTX 4090 full benchmarks failed. (${{ env.ACTION_RUN_URL }})"
+
+  cuda-core-crypto-benchmarks:
+    name: Cuda core crypto benchmarks  (RTX 4090)
+    if: ${{ github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || contains(github.event.label.name, '4090_bench') }}
+    concurrency:
+      group: ${{ github.workflow }}_${{ github.ref }}_cuda_core_crypto_bench
+      cancel-in-progress: true
+      needs: ${{ github.workflow }}_${{ github.ref }}_cuda_integer_bench
```

Comment:
Does the `needs` directive in `concurrency` works ? :thinking: 
In any case, I thinks it's more understandable if `needs` is put right after the `if` (one indentation-level higher).
By using `needs: cuda-integer-benchmarks`, you'll ensure that all sub-jobs in the first step are done.